droidlet/dashboard/web/src/components/Interact/Question.js

```diff
@@ -337,36 +277,288 @@ class Question extends Component {
+          }
+        );
+        locationRef =
+          "'" + antecedent.join(" ").replace(/  +/g, " ").trim() + "'";
+      }
+    }
+    return locationRef;
+  }
+
+  evalCommandPerception() {
+    //                 Reference Object Exists in dictionary ?
+    //                         /   \
+    //                       Yes    No -> Other Error
+    //                        |
+    //             Found Location in Memory?
+    //                      /   \
+    //                    Yes    No -> Perception Error
+    let ref_object = false;
+    let reference_object_description = null;
+    // Check if reference object exists in the dictionary anywhere
+    if (this.state.action_dict) {
+      if (this.state.action_dict["dialogue_type"] == "HUMAN_GIVE_COMMAND") {
+        // also implement for get and put memory
+        for (const action of this.state.action_dict.action_sequence) {
+          ref_object = this.check_reference_object_in_action_dict(action);
+        }
+      }
+
+      // If yes, find reference object description.
+      if (ref_object == true) {
+        const action_dict = this.state.action_dict.action_sequence[0];
+        // Check for location at top level and extract the reference text
+        let considered_action_dict = null;
+        if ("location" in action_dict) {
+          considered_action_dict = action_dict.location;
+        } else if ("reference_object" in action_dict) {
+          considered_action_dict = action_dict;
+        }
+
+        if (considered_action_dict) {
+          reference_object_description = this.extractLocationRef(
+            considered_action_dict
+          );
+        }
+      }
+      // If no reference object description found no perception error.
+    } else {
+      console.log("no action dictionary found ..."); // Shouldn't happen....
+    }
+    this.setState({
+      reference_object_description: reference_object_description,
+    });
+
+    // else {
+    //   // shouldn't happen
+    //   return (
+    //     <div>
+    //       <h3> Thanks! Press to continue.</h3>
+    //       <Button
+    //         variant="contained"
+    //         color="primary"
+    //         onClick={() => this.props.goToMessage()}
+    //       >
+    //         Done
+    //       </Button>
+    //     </div>
+    //   );
+    // }
+  }
+
+  renderVisionQuestion() {
+    //        Is 'This' the Location Ref?
+    //                  /   \
+    // Other Error <- Yes    No -> Perception Error
+    let reference_object_description = this.state.reference_object_description;
+    if (!reference_object_description) {
+      // Not perception error.
+      this.setState({ view: 6 });
+      return;
+    }
+    return (
+      <div className="question">
+        <h3>
+          Okay, looks like I understood your command. I was looking for an
+          object of interest called : {reference_object_description}. Here's
+          what I think it is. Does that look right ?
```

Comment:
I think we should explicitly tell people to look for the flashing object.

Reply:
Thanks, yes will do now. Flashing wasn't yet implemented